apps/system/js/notifications.js

```diff
@@ -396,6 +409,16 @@ var NotificationScreen = {
       this.container.querySelector('.priority-notifications') :
       this.container.querySelector('.other-notifications');
 
+    /* The auto value falls back to the current
+     * direction of the document, i.e ltr or rtl
+     * to prevent misplaced child items of the
+     * notification structure when mixed content
+     * is used.
+     */
```

Comment:
I don't understand this comment very well. I think your explanation to Alive [1] was more easily understandable. Let's do two things here. First, let's be explicit about what cases we need to work around in the code. ie. let's have something like:
```
var dir = (detail.bidi === 'auto' || typeof detail.bidi === 'undefined') ?
  document.documentElement.dir : detail.bidi;
```
Then, in the comment let's say something like: `If dir "auto" was specified by the notification, use document direction instead because dir="auto" does xxxxx and we need yyyyyyy.`

1.) https://github.com/mozilla-b2g/gaia/pull/27020#discussion_r22347661